native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/states/http2/SendingRstFrame.java

```diff
@@ -100,16 +104,29 @@ public void handleStreamTimeout(OutboundMsgHolder outboundMsgHolder, boolean ser
                                     int streamId) throws Http2Exception {
         if (!serverPush) {
             outboundMsgHolder.getResponseFuture().notifyHttpListener(
-                    new EndpointTimeOutException(IDLE_TIMEOUT_TRIGGERED_WHILE_WRITING_OUTBOUND_REQUEST_HEADERS,
+                    new EndpointTimeOutException(IDLE_TIMEOUT_TRIGGERED_WHILE_SENDING_RST_STREAM,
```

Comment:
Why did we change this error? This is where the client gets timeout right?

Reply:
Yes. But the current state is sending rst stream. But the message says the state is sending headers

Reply:
Got it 👍🏼